bin/spark-sql.cmd

```diff
@@ -0,0 +1,25 @@
+@echo off
+
+rem
+rem Licensed to the Apache Software Foundation (ASF) under one or more
+rem contributor license agreements.  See the NOTICE file distributed with
+rem this work for additional information regarding copyright ownership.
+rem The ASF licenses this file to You under the Apache License, Version 2.0
+rem (the "License"); you may not use this file except in compliance with
+rem the License.  You may obtain a copy of the License at
+rem
+rem    http://www.apache.org/licenses/LICENSE-2.0
+rem
+rem Unless required by applicable law or agreed to in writing, software
+rem distributed under the License is distributed on an "AS IS" BASIS,
+rem WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+rem See the License for the specific language governing permissions and
+rem limitations under the License.
+rem
+
+rem This is the entry point for running SparkSQL. To avoid polluting the
+rem environment, it just launches a new cmd to do the real work.
+
+rem The outermost quotes are used to prevent Windows command line parse error
+rem when there are some quotes in parameters, see SPARK-21877.
+cmd /V /E /C ""%~dp0spark-sql2.cmd" %*"
```

Comment:
Separate script is required. Otherwise, it will actually set the environment variable, `SPARK_HOME`. See SPARK-3943 and I also manually tested.

Reply:
just for curious, why does the shell script not need a seprated script?

Reply:
AFAIK, `export` in bash affects its subprocesses so it doesn't affect the parent. I somehow knew this but .. haven't thought deeply before. I think this is just a difference between OSes, maybe missing concept of forking on Windows.

Reply:
BTW, to be honest, I didn't know my little Windows env dev experience a long ago could be helpful to Spark at the end :).